From eb1784185732c03a4324c141e151dc67b2e47f75 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@ew.tq-group.com>
Date: Fri, 28 Feb 2025 17:18:09 +0100
Subject: [PATCH 06/29] arm: dts: imx8mp: [HACK] add nodes to enable usage of
 upstream dt

Some nodes for audio / HDMI are missing to compile with upstream trees.
Add the nodes only with reg property and status set to disabled.

Signed-off-by: Markus Niebel <Markus.Niebel@ew.tq-group.com>
---
 arch/arm/dts/imx8mp.dtsi | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm/dts/imx8mp.dtsi b/arch/arm/dts/imx8mp.dtsi
index 1449fb2063e..6e4fb2bffac 100644
--- a/arch/arm/dts/imx8mp.dtsi
+++ b/arch/arm/dts/imx8mp.dtsi
@@ -1394,6 +1394,18 @@
 					status = "disabled";
 				};
 
+				aud2htx: aud2htx@30cb0000 {
+					reg = <0x30cb0000 0x10000>;
+					status = "disabled";
+				};
+
+				xcvr: xcvr@30cc0000 {
+					reg = <0x30cc0000 0x800>,
+					      <0x30cc0800 0x400>,
+					      <0x30cc0c00 0x080>,
+					      <0x30cc0e00 0x080>;
+					status = "disabled";
+				};
 			};
 
 			sdma3: dma-controller@30e00000 {
@@ -1508,6 +1520,11 @@
 				};
 			};
 
+			lcdif2: display-controller@32e90000 {
+				reg = <0x32e90000 0x10000>;
+				status = "disabled";
+			};
+
 			media_blk_ctrl: blk-ctrl@32ec0000 {
 				compatible = "fsl,imx8mp-media-blk-ctrl",
 					     "syscon";
@@ -1576,6 +1593,26 @@
 				#power-domain-cells = <1>;
 				#clock-cells = <0>;
 			};
+
+			hdmi_pvi: display-bridge@32fc4000 {
+				reg = <0x32fc4000 0x1000>;
+				status = "disabled";
+			};
+
+			lcdif3: display-controller@32fc6000 {
+				reg = <0x32fc6000 0x1000>;
+				status = "disabled";
+			};
+
+			hdmi_tx: hdmi@32fd8000 {
+				reg = <0x32fd8000 0x7eff>;
+				status = "disabled";
+			};
+
+			hdmi_tx_phy: phy@32fdff00 {
+				reg = <0x32fdff00 0x100>;
+				status = "disabled";
+			};
 		};
 
 		dma_apbh: dma-apbh@33000000 {
-- 
2.34.1

